pkg/k8shandler/status.go

```diff
@@ -47,15 +45,55 @@ func updateNodeStatus(node *nodeState, dpl *v1alpha1.ElasticsearchStatus) {
 	if node.Actual.StatefulSet != nil {
 		nodeStatus.StatefulSetName = node.Actual.StatefulSet.Name
 	}
+
+	if node.Desired.Roles != nil {
+		nodeStatus.Roles = node.Desired.Roles
+	}
 	dpl.Nodes = append(dpl.Nodes, nodeStatus)
 }
 
-func AddStatusMessage(message string, dpl *v1alpha1.Elasticsearch) error {
-	dpl.Status.Message = message
-	err := sdk.Update(dpl)
+func clusterHealth(dpl *v1alpha1.Elasticsearch) string {
+	pods, err := listPods(dpl.Name, dpl.Namespace)
 	if err != nil {
-		return fmt.Errorf("failed to update Elasticsearch status: %v", err)
+		return healthUnknown
 	}
 
-	return nil
+	// no elasticsearch pods were found
+	if len(pods.Items) == 0 {
+		return healthUnknown
+	}
+
+	// use arbitrary pod
+	pod := pods.Items[0]
+
+	config := &ExecConfig{
+		pod:            &pod,
+		containerName:  "elasticsearch",
+		command:        []string{"es_util", "--query=_cluster/health?pretty=true"},
+		kubeConfigPath: LookupEnvWithDefault("KUBERNETES_CONFIG", "/etc/origin/master/admin.kubeconfig"),
+		masterURL:      "https://kubernetes.default.svc",
```

Comment:
I think this is also available via a well known env var that is injected into the container.  Do we prefer this name over using the value in the env var?  I don't know the right answer, I'm just asking

Reply:
I see them as interchangeable - the above is the domain name of k8s service and what we get from the environment is IP address for that service. That IP address should be always the same (pod IPs change, svcs should be constant), so we shouldn't have to worry about DNS caching. That's why I believe that either works the same.

Reply:
I think in general it is safer for us to use a service end point over a pod address in this case.